main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,131 @@
+      List<FastTravelBetweenStopsNotice> noticesForTrip = checkSpeedAlongTrip(entry);
+      for (FastTravelBetweenStopsNotice notice : noticesForTrip) {
+        noticeContainer.addValidationNotice(notice);
+      }
+    }
+  }
+
+  /**
+   * Calculates the travel speed between {@code GtfsStopTime}. Generates a {@code
+   * FastTravelBetweenStopsNotice} if the travel speed between {@code GtfsStopTimes} is greater than
+   * 150 km/h.
+   *
+   * @param tripStopTimes the list of {@code GtfsStopTimes} sorted by stop_times.stop_sequence for a
+   *     {@code GtfsTrip}
+   * @return the list of {@code FastTravelBetweenStopsNotice} generated when checking speed travel
+   *     along the given trip
+   */
+  private List<FastTravelBetweenStopsNotice> checkSpeedAlongTrip(
+      Entry<String, List<GtfsStopTime>> tripStopTimes) {
+    var previousStopsData =
+        new Object() {
+          final List<Integer> accumulatedStopSequence = new ArrayList<>();
+          GtfsTime departureTime = null;
+          double latitude;
+          double longitude;
+          // used to accumulate distance between stops with same arrival and departure
+          // times
+          int accumulatedDistanceMeter = 0;
+        };
+    List<FastTravelBetweenStopsNotice> notices = new ArrayList<>();
+    for (GtfsStopTime stopTime : tripStopTimes.getValue()) { // prepare data for current iteration
+      GtfsStop currentStop = stopTable.byStopId(stopTime.stopId());
+      double currentStopLat = currentStop.stopLat();
+      double currentStopLon = currentStop.stopLon();
+      GtfsTime currentArrivalTime = stopTime.arrivalTime();
+      double distanceFromPreviousStopMeter =
+          GeospatialUtil.distanceInMeterBetween(
+              previousStopsData.latitude,
+              previousStopsData.longitude,
+              currentStopLat,
+              currentStopLon);
+      boolean sameArrivalAndDeparture = false;
+      if (previousStopsData.departureTime != null && currentArrivalTime != null) {
+        sameArrivalAndDeparture = currentArrivalTime.equals(previousStopsData.departureTime);
+        if (!sameArrivalAndDeparture) {
+          int durationSecond =
+              currentArrivalTime.getSecondsSinceMidnight()
+                  - previousStopsData.departureTime.getSecondsSinceMidnight();
+          double distanceMeter =
+              distanceFromPreviousStopMeter + previousStopsData.accumulatedDistanceMeter;
+          double speedMeterPerSecond = distanceMeter / durationSecond;
+          if (speedMeterPerSecond > 42) { // roughly 150 km per hour. Put it in default parameters
+            previousStopsData.accumulatedStopSequence.add(stopTime.stopSequence());
+            notices.add(
+                new FastTravelBetweenStopsNotice(
+                    tripStopTimes.getKey(),
+                    speedMeterPerSecond * METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR,
+                    new ArrayList<>(previousStopsData.accumulatedStopSequence)));
```

Comment:
I don't think the stop sequence list really provides value here? Unless I'm missing something. I'd be inclined to omit this to avoid another ArrayList allocation.

Reply:
This stop sequence lets us know between which stop times the travel is too fast. If we omit it, we would just know that  -for example- for trip `t0` the speed is incorrect which would not allow us to traceback the wrong stop times.

Reply:
Couldn't you do the same with one or two stop sequence values without the list though? I'm not seeing why you would need more than two values at the most (the starting and ending stop sequence) to define where the travel is too fast. So you could say "Too fast travel starting at X" or "Too fast travel between X and Y", but I don't see why you'd need a third value.

Reply:
Totally true! Did not realize it that way. Modifying this to reflect your thoughts. Thanks @barbeau  :)